pandas/tests/reshape/merge/test_multi.py

```diff
@@ -460,6 +460,38 @@ def test_merge_na_keys(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_merge_na_datetime_keys_empty_df(self):
+        data = [
+            [pd.Timestamp("1950-01-01"), "A", 1.5],
+            [pd.Timestamp("1950-01-02"), "B", 1.5],
+            [pd.Timestamp("1950-01-03"), "B", 1.5],
+            [pd.Timestamp("1950-01-04"), "B", np.nan],
+            [pd.Timestamp("1950-01-05"), "B", 4.0],
+            [pd.Timestamp("1950-01-06"), "C", 4.0],
+            [pd.Timestamp("1950-01-07"), "C", np.nan],
+            [pd.Timestamp("1950-01-08"), "C", 3.0],
+            [pd.Timestamp("1950-01-09"), "C", 4.0],
+        ]
+
+        frame = DataFrame(data, columns=["date", "panel", "data"]).set_index(
+            ["date", "panel"]
+        )
+
+        other_data = []
+        other = DataFrame(other_data, columns=["date", "panel", "state"]).set_index(
+            ["date", "panel"]
+        )
+
+        expected = DataFrame([], columns=["date", "panel", "data", "state"])
+        expected[["date", "panel", "data"]] = frame.reset_index()[
+            ["date", "panel", "data"]
+        ]
+        expected = expected.set_index(["date", "panel"])
```

Comment:
Can you write these tests without using set_index or reset_index (building data explicitly using constructors)? Helps isolate the testing to only merge and make things easier to understand. Also could make the original data smaller.